### PR TITLE
Polish API and documentation for lang and expr functions

### PR DESCRIPTION
**Pull request by lionel-**

### CODE REVIEW

File: R/expr-lang.R [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #' Create a language call by "hand"
 #'
-#' @param .fn Function to call. Must be a callable object: a string,
-#'   symbol, call, or a function.
+#' @param .fn Function to call. Must be a callable object: symbol,
```
Comment (hadley) [inline R/expr-lang.R]: A string still is ok though? (And I think it should be for convenience)

File: R/expr-lang.R [hunk 2]
```diff
@@ -40,8 +40,35 @@ lang <- function(.fn, ..., .ns = NULL) {
...
+#' is not legal to manually set language heads to strings.
+#'
+#' @param x An object to test.
+#' @export
+#' @examples
+#' # Symbolic objects and functions are callable:
+#' is_callable(quote(foo))
+#' is_callable(base::identity)
+#'
+#' # You can safely set the head of a language node to a callable
```
Comment (hadley) [inline R/expr-lang.R]: I don't get what you're trying to say here

File: R/expr-lang.R [hunk 3]
```diff
@@ -203,17 +231,17 @@ is_binary_lang <- function(x, name = NULL, ns = NULL) {
 #' # You can also modify quosures inplace:
 #' f <- ~matrix(bar)
 #' lang_modify(f, quote(foo))
-lang_modify <- function(.call = caller_frame(), ..., .standardise = FALSE) {
+lang_modify <- function(.lang = caller_frame(), ..., .standardise = FALSE) {
```
Comment (hadley) [inline R/expr-lang.R]: I wonder if these have diverged sufficiently from the original function that the first argument should no longer have a default.

File: R/expr-lang.R [hunk 4]
```diff
@@ -326,41 +356,43 @@ lang_fn <- function(call = caller_frame()) {
...
-#' Return the head or tail of a call object.
+#' Return the head or tail of a call object
 #'
 #' @description
 #'
-#' Internally, calls are structured as a tree of expressions (see
-#' [switch_lang()] and [pairlist] documentation pages). A `language`
-#' object is the top level node of the tree. `lang_head()` and
-#' `lang_tail()` allow you to retrieve the node components.
+#' Internally, calls (`language` objects) are structured as a tree of
```
Comment (hadley) [inline R/expr-lang.R]: This seems inconsistent with how we use "language" elsewhere. I think you need to somehow qualify this is what base R calls these things?

File: R/expr-lang.R [hunk 5]
```diff
@@ -326,41 +356,43 @@ lang_fn <- function(call = caller_frame()) {
...
+#'   checks that the head is a symbol and converts it to a string).
 #'
 #' * The second component of the tree node contains the arguments. The
-#'   type of arguments is _pairlist_. Pairlists are actually
+#'   type of arguments is _pairlist_. Pairlists are structurally
 #'   equivalent to `language` objects (calls), they just have a
-#'   different name. `lang_tail()` returns the pairlist of arguments
-#'   ([lang_args()] returns the same object converted to a regular
-#'   list).
+#'   different type name. `lang_tail()` returns the pairlist of
```
Comment (hadley) [inline R/expr-lang.R]: Remove name?

File: R/expr-lang.R (hunk 1 above)
Comment (lionel-) [inline R/expr-lang.R]: oops yes it is still ok there

File: R/expr-lang.R (hunk 2 above)
Comment (lionel-) [inline R/expr-lang.R]: `mut_node_car()` lets you put anything in a call and that can lead to crashes in some circumstances. `is_callable()` is a predicate that checks the node head is safely callable.

**Comment by lionel-** [comment]
I will fix the lang vs call thing tomorrow

**Pull request merged**